internal/resources/resource_volume.go

```diff
@@ -260,22 +264,71 @@ func resourceQuatrroVolumeRead(d *schema.ResourceData, meta interface{}) (err er
 	}
 	d.Set(vState, volume.State)
 	d.Set(vStatus, volume.Status)
+
 	return nil
 }
 
-func resourceQuatrroVolumeUpdate(d *schema.ResourceData, meta interface{}) (err error) {
+func resourceMetalVolumeUpdate(d *schema.ResourceData, meta interface{}) (err error) {
 	defer func() {
 		var nErr = rest.GenericOpenAPIError{}
 		if errors.As(err, &nErr) {
 			err = fmt.Errorf("failed to update volume %s: %w", strings.Trim(nErr.Message(), "\n "), err)
 
 		}
 	}()
+
 	//@TODO - or not....?
-	return resourceQuatrroVolumeRead(d, meta)
+	return resourceMetalVolumeRead(d, meta)
+}
+
+// deleteVAsForVolume deletes all attachments for specified volume.
+func deleteVAsForVolume(p *configuration.Config, volID string) error {
+	ctx := p.GetContext()
+
+	// Get all attachments
+	vas, _, err := p.Client.VolumeAttachmentsApi.List(ctx)
+	if err != nil {
+		return fmt.Errorf("error listing volume attachments: %w", err)
+	}
+
+	// Initiate attachments deletion for this volume
+	for _, va := range vas {
+		if va.VolumeID == volID {
+			_, err = p.Client.VolumeAttachmentsApi.Delete(ctx, va.ID)
+			if err != nil {
+				return fmt.Errorf("error deleting volume attachment %s: %w", va.ID, err)
```

Comment:
We've adopted standard on err messages to not repeat the "error" since we know this is an error, so keep it simple.  This help especially when errors are wrapped w/ %w -
```suggestion
				return fmt.Errorf("delete volume attachment %s: %w", va.ID, err)
```

Reply:
Done.